### ISSUE TITLE
v3.1.0rc2: Fetching unknown toots via search field returns Error 500

### ISSUE THREAD

**Issue by abochmann** [tootsuite/mastodon#13022]
Ever since upgrading my Mastodon instance to v3.1.0rc2, I'm not able to retrieve unknown posts from other instances by dropping their URL into the search field. I'm using this functionality relatively frequently (single-user instance).

The web UI returns an "500 Internal Server Error" message, but I can not find any hints about possible causes in the logs.

This may be connected to issue #13011 

### Expected behaviour

Using search with the https URL of a public remote toot loads the previously unknown post, adds it to the local database, and displays the contents as search result.
 
### Actual behaviour

The Mastodon web UI shows an "500 Internal Server Error"

### Steps to reproduce the problem

Find a remote toot that's not known to your instance.
Use the URL in your search field to attempt to load the toot.

### Specifications

v3.1.0rc2

**Comment by ThibG** [comment]
If you get “500 Internal Server Error”, there should be more information in puma/mastodon-web's logs (journalctl -u mastodon-web, depending on how you installed it)

**Comment by abochmann** [comment]
Thanks. The only log entries I can see in conjunction with such a failed request, based on time, is a media processing error (avatar, possibly?):

`[4e495f36-e5e1-4477-994a-304a9f13037e] Command :: file -b --mime '/tmp/8d777f385d3dfec8815d20f7496026dc20200202-554-11uhu7u'
[4e495f36-e5e1-4477-994a-304a9f13037e] [paperclip] Trying to link /tmp/8d777f385d3dfec8815d20f7496026dc20200202-554-11uhu7u to /tmp/8870409710f8e4fe532916b589bcd71e20200202-554-qgon5e.jpeg
[4e495f36-e5e1-4477-994a-304a9f13037e] Command :: file -b --mime '/tmp/8870409710f8e4fe532916b589bcd71e20200202-554-qgon5e.jpeg'
[4e495f36-e5e1-4477-994a-304a9f13037e] [paperclip] Trying to link /tmp/8d777f385d3dfec8815d20f7496026dc20200202-554-11uhu7u to /tmp/cdb1a64a2ddb5883599a5d26314e064220200202-554-1mcvqas
[4e495f36-e5e1-4477-994a-304a9f13037e] [paperclip] Link failed with No such file or directory @ rb_file_s_link - (/tmp/8d777f385d3dfec8815d20f7496026dc20200202-554-11uhu7u, /tmp/cdb1a64a2ddb5883599a5d26314e064220200202-554-1mcvqas); copying link /tmp/8d777f385d3dfec8815d20f7496026dc20200202-554-11uhu7u to /tmp/cdb1a64a2ddb5883599a5d26314e064220200202-554-1mcvqas
[4e495f36-e5e1-4477-994a-304a9f13037e] method=GET path=/api/v2/search format=html controller=Api::V2::SearchController action=index status=500 error='Errno::ENOENT: No such file or directory - No such file or directory @ rb_sysopen - /tmp/8d777f385d3dfec8815d20f7496026dc20200202-554-11uhu7u' duration=278.77 view=0.00 db=5.68
[4e495f36-e5e1-4477-994a-304a9f13037e]
[4e495f36-e5e1-4477-994a-304a9f13037e] Errno::ENOENT (No such file or directory - No such file or directory @ rb_sysopen - /tmp/8d777f385d3dfec8815d20f7496026dc20200202-554-11uhu7u):
[4e495f36-e5e1-4477-994a-304a9f13037e]
[4e495f36-e5e1-4477-994a-304a9f13037e] /home/mastodon/.rbenv/versions/2.6.5/lib/ruby/2.6.0/fileutils.rb:1385:in ``initialize'
`

**Comment by ThibG** [comment]
It looks like you may be using an incorrect version of Paperclip

How did you install Mastodon? Is your Gemfile.lock modified?

**Comment by abochmann** [comment]
Hrm. According to a _git status_, it is (together with package.json and yarn.lock). This has happened before, as I remember now 🙄
I'll try to fix it ASAP and report back.

For updates, I do the usual git pull / git checkout <version> / bundle install / yarn install, so I don't currently know where I might have gone wrong with that.

**Comment by abochmann** [comment]
Ok, paperclip was at the correct version, but a couple of others weren't (json-ld-preloaded seems like a likely culprit).

Sorry for the noise, I'll close the issue.

Is it possible that running a _bundle update_ could lead to an effect like this? Seems that has snuck in somewhere from cutting and pasting lines from shell history...

**Comment by ThibG** [comment]
Yes, `bundle update` is the culprit.
I'd recommend running `bundle config set frozen true` and `bundle config set deployment true` so that it'll complain in those cases and not let you update the Gemfile.lock

**Comment by abochmann** [comment]
👍 Thank you again for your help!

**Comment by EasternPA** [comment]
Sorry to necropost here, but I'm having this exact issue. I revisited some old build instructions I wrote a couple of years ago, and the `bundle exec rake secret` step now fails with:

`rake aborted!
Don't know how to build task 'secret' (See the list of available tasks with `rake --tasks`)`

I went ahead anyway, and now I'm facing the error described in this issue. Every time I copy/paste a Mastodon URL into the search field, I get a `500` slide in and out at the bottom left. Can I just use any value as the PAPERCLIP secret or did the rake secret command store the output value somewhere? I usually use `date | md5sum` for a good password, so I'll just use that if that's all it needs.
